client/components/command-pallette/README.md

```diff
@@ -0,0 +1,68 @@
+
+The command palette is inspired by [Core](https://github.com/WordPress/gutenberg/blob/trunk/packages/commands/README.md) implementation used in the Gutenberg editor. We use the same CSS and similar markup, the commands share the same basic props that we extended to support the following differences:
+
+- The Calypso version supports multisite commands with the sites as second step commands.
+- Users can go back to root commands with the `esc` and `backspace` keys, or the back button.
+- We change the search placeholder to inform the user about the current command.
+- Besides of icons, we also support images for the commands.
+- We support two lines of text for the commands.
+
+## Add a command
+
+In order to add a new command to the palette, you can include the command object in `useCommandsArrayWpcom` or create a new hook and include it to `useCommandPallette` hooks.
+
+## Command properties
+
+Every command has a set of properties that can be used to customize its behavior. The following properties are available:
+
+- `name`: Type `string`, used as identifier for the command. It's expected to be unique and `camelCase`. It's not visible for the user.
+- `label`: Type `string`, used as the label for the command. Visible in the command palette.
+- `subLabel`?: Type `string`, used as a second line for the command. Visible in the command palette.
+- `searchLabel`?: Type `string`, used to match the user search. It's not visible for the user. We recommend to use `_x()` to add context to the translators.
+- `callback`: Type `function`, used to execute the command when the user clicks on it or presses enter on that command.
+- `context`?: Type `string[]`, a list of URL paths for which the command will have priority when displayed.
+- `icon`?: Type `JSX.Element`, used to display an icon for the command. It's visible in the command palette.
+- `image`?: Type `JSX.Element`, used to display an image for the command, instead of an icon. It's visible in the command palette.
+- `siteFunctions`?: Type `object`, used for nested commands that need to execute a function when a site is selected in the command palette as a second step.
+
+## Best practices for defining a command
+
+Please consider carefully before adding a new command to the palette. We aim to avoid incorporating commands that are infrequently used and unlikely to be removed in the future.
+
+- Command labels should be short and descriptive and use sentence case. For example, "Add new site" instead of "Add New Site".
+- Commands should start with a verb. For example, "Add new site" instead of "New site".
+- Picking the correct verb is important. `Open` is the more generic verb, if the page has a list of elements, we can use `View`, if the user is expected to take an action we can use `Manage` or `Change`. We prefer specific verbs over generic ones.
```

Comment:
Another aspect is to stick with terminology used elsewhere in the interace, and use aliases for synonyms.

Examples:
- "Import a site" because we have Tools → Import, and alias "migrate a site".
- "Add a post" because "add" is used in WP instead of "write".

Reply:
Done here https://github.com/Automattic/wp-calypso/pull/84591/commits/13158f7b2c189c0993e70e8ec224cd08cb89634c